aspnetcore/release-notes/aspnetcore-7.0.md

```diff
@@ -209,6 +209,202 @@ SignalR hub methods now support injecting services through dependency injection
+
+### Virtualization enhancements
+
+Virtualization enhancements in .NET 7:
+
+* The `Virtualize` component supports using the document itself as the scroll root, as an alternative to having some other element with `overflow-y: scroll` applied.
+* If the `Virtualize` component is placed inside an element that requires a specific child tag name, `SpacerElement` allows you to obtain or set the virtualization spacer tag name.
+
+For more information, see the following sections of the *Virtualization* article:
+
+* [Root-level virtualization](xref:blazor/components/virtualization?view=aspnetcore-7.0#root-level-virtualization)
+* [Control the spacer element tag name](xref:blazor/components/virtualization?view=aspnetcore-7.0#control-the-spacer-element-tag-name)
+
+### `MouseEventArgs` updates
+
+`MovementX` and `MovementY` have been added to `MouseEventArgs`.
+
+For more information, see <xref:blazor/components/event-handling?view=aspnetcore-7.0#built-in-event-arguments>.
+
+### New Blazor loading page
+
+The Blazor WebAssembly project template has a new loading UI that shows the progress of loading the app.
+
+For more information, see <xref:blazor/fundamentals/startup?view=aspnetcore-7.0#loading-progress-indicators>.
+
+### Improved diagnostics for authentication in Blazor WebAssembly
+
+To help diagnose authentication issues in Blazor WebAssembly apps, detailed logging is available.
+
+For more information, see <xref:blazor/fundamentals/logging?view=aspnetcore-7.0#blazor-webassembly-authentication-logging>.
+
+<!--
+
+    IN PROGRESS
+
+### .NET JavaScript interop on WebAssembly
+
+.NET JavaScript `[JSImport]`/`[JSExport]` interop API is a new low-level mechanism for using .NET in Blazor WebAssembly and JavaScript-based apps. With this new JavaScript interop capability, you can invoke .NET code from JavaScript using the .NET WebAssembly runtime and call into JavaScript functionality from .NET without any dependency on the Blazor UI component model.
+
+For more information, see the following articles:
+
+* <xref:blazor/js-interop/import-export-interop>: Pertains only to Blazor WebAssebmly apps.
+* <xref:client-side/import-export-interop>: Pertains only to JavaScript apps that don't depend on the Blazor UI component model.
+
+Under consideration for docs ...
+
+[Blog Post (WebAssembly multithreading experimental)](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-7-rc-2/#webassembly-multithreading-experimental)
+
+-->
+
+### Conditional registration of the authentication state provider
+
+Prior to the release of .NET 7, `AuthenticationStateProvider` was registered in the service container with `AddScoped`. This made it difficult to debug apps, as it forced a specific order of service registrations when providing a custom implementation. Due to internal framework changes over time, it's no longer necessary to register `AuthenticationStateProvider` with `AddScoped`:
+
+```diff
```

Comment:
Is this diff showing what changed in the product code, or what should change in user code? It looks like the former, which is a bit weird. Maybe this whole section is just a bug fix that should just be removed?

Reply:
Pretty much user code. It's just showing that for all of our examples, which are standardized on the "`ExternalAuthStateProvider`" naming, they've all changed for .NET 7 docs like this.

We told folks for a *long time* ... from the beginning of Blazor security docs ... to register with `AddScoped`. I thought it would be good to mention this for the reason that Javier stated.

Reply:
I have a feeling that when he said ...

> This made it difficult to debug apps

He means that funky error message(s) aren't super clear that the problem is the order of service registrations, but I'm just guessing about that.

Reply:
I can add something like ....

> In user code, make the following change to the authentication state provider service registration: